tokens/token-2022/transfer-hook/token-stop/anchor/programs/token-stop/src/lib.rs

```diff
@@ -0,0 +1,160 @@
+use anchor_lang::prelude::*;
+
+use anchor_spl::{
+    token_interface::{
+        spl_token_metadata_interface::state::TokenMetadata, Mint,
+        Token2022, token_metadata_update_field, TokenMetadataUpdateField
+    },
+    token_2022::{
+        spl_token_2022::{
+            extension::{
+                BaseStateWithExtensions,
+                StateWithExtensions,
+            },
+            state::Mint as Token2022Mint
+        },
+        ID as TOKEN_2022_PROGRAM_ID,
+    },
+};
+
+use spl_tlv_account_resolution::{account::ExtraAccountMeta, state::ExtraAccountMetaList};
+use spl_transfer_hook_interface::instruction::ExecuteInstruction;
+
+declare_id!("Hy7xLUkd52wkE2xNYvNVYmmrLeqfkoqnw6sYK3cqbNqf");
+
+#[error_code]
+pub enum TransferStopError {
+    #[msg("The transfers have been disabled")]
+    TransfersDisabled,
+}
+
+#[program]
+pub mod token_stop {
+    use anchor_spl::token_interface::spl_token_metadata_interface::state::Field;
+
+    use super::*;
+
+    pub fn stop_transfer(ctx: Context<StopTransfer>) -> Result<()> {
```

Comment:
Looks like anyone could trigger the stop or resume the transfer. A nice addition would be to save the authority also in the token meta data or add an admin key check or use the mint authority as admin .